iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -109,20 +110,22 @@ public static boolean isCause(Class<? extends Throwable> possibleExceptionCause,
         return possibleExceptionCause.isInstance(exceptionToSearch) || (exceptionToSearch != null && isCause(possibleExceptionCause, exceptionToSearch.getCause()));
     }
 
-    public synchronized static Device addDeviceWithRetry(RegistryManager registryManager, Device device) throws IotHubException, IOException, InterruptedException
```

Comment:
This method being synchronized meant that only one thread can enter this static method at a time. Because of this, tests have been failing lately just by getting stuck waiting in line to enter this method. In most cases, this method is called during each test setup, so it is called a lot. 

By removing the synchronization, the SDK test side doesn't slow itself down in making these requests, but IoT Hub often responds slower because of it, or responds with a throttle. The added throttleResistantTestRule covers the latter, and using infinite read timeouts covers the former. As a result, our tests scale better if we ever upgrade to an S2 for testing, but still work against an S1